src/rules/noDuplicateImportsRule.ts

```diff
@@ -44,13 +44,38 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 }
 
-function walk(ctx: Lint.WalkContext<void>) {
-    const seen = new Set<string>();
-    for (const {text, parent} of findImports(ctx.sourceFile, ImportKind.ImportDeclaration)) {
-        if (seen.has(text)) {
-            ctx.addFailureAtNode(parent!, Rule.FAILURE_STRING(text));
-        } else {
+function walk(ctx: Lint.WalkContext<void>): void {
+    walkWorker(ctx, ctx.sourceFile.statements, new Set());
+}
+
+function walkWorker(ctx: Lint.WalkContext<void>, statements: ReadonlyArray<ts.Statement>, seen: Set<string>): void {
+    for (const statement of statements) {
+        const im = tryGetImportSpecifier(statement);
+        if (im !== undefined && isLiteralExpression(im)) {
+            const { text } = im;
+            if (seen.has(text)) {
+                ctx.addFailureAtNode(im.parent!, Rule.FAILURE_STRING(text));
+            }
             seen.add(text);
         }
+
+        if (isModuleDeclaration(statement) && statement.body !== undefined && isModuleBlock(statement.body)) {
```

Comment:
you could add an additional check here if `statement.name.kind === ts.SyntaxKind.StringLiteral`, because namespaces cannot contain import statements.
and then you probably don't need the `isModuleBlock` check.